src/gennylib/src/Orchestrator.cpp

```diff
@@ -136,8 +136,12 @@ bool Orchestrator::awaitPhaseEnd(bool block, int removeTokens) {
     // compare with >= rather than ==.
 
     if (_currentTokens <= 0) {
+        const auto currentPhase = _current;
         ++_current;
-        BOOST_LOG_TRIVIAL(info) << "Ended phase " << (this->_current - 1);
+        BOOST_LOG_TRIVIAL(info) << "Ended phase " << currentPhase;
+        for (auto&& cb : _postPhaseHooks) {
+            cb(this, currentPhase);
```

Comment:
Do you want the next phase number for the call back, or the current? could you comment that somewhere?

Reply:
Looking below I think you want to call this line before incrementing current on line 140.

Reply:
All good now. Thanks. I still don't think the old code works, but that's no longer relevant :)